src/hutch_bunny/core/solvers/rule_query_builders.py

```diff
@@ -0,0 +1,469 @@
+
+        Args:
+            left_value_time (str): Left-side time bound in months as a numeric string,
+                or empty string if unused.
+            right_value_time (str): Right-side time bound in months as a numeric string,
+                or empty string if unused.
+
+        Returns:
+            OMOPRuleQueryBuilder: The current instance with updated query filters.
+
+        Notes:
+        - This method assumes the input strings have already been parsed and
+          validated (e.g., "|6" converted to "6") before being passed in.
+        - The time values represent months relative to the current date.
+        """
+
+        if not left_value_time and not right_value_time:
+            raise ValueError(
+                "Temporal constraint requires exactly one time value. "
+                "Both left_value_time and right_value_time are empty."
+            )
+        
+        if left_value_time and right_value_time:
+            raise ValueError(
+                "Temporal constraint requires exactly one time value. "
+                f"Both values were provided: left='{left_value_time}', right='{right_value_time}'. "
+                "One must be an empty string."
+            )
+
+        if left_value_time == "":
+            time_value_supplied = right_value_time
+        else:
+            time_value_supplied = left_value_time
+
+        time_to_use = int(time_value_supplied)
+        time_to_use = time_to_use * -1
+
+        today_date = datetime.now()
+        relative_date = today_date + relativedelta(months=time_to_use)
+
+        if left_value_time == "":
+            self.measurement_query = self.measurement_query.where(
+                Measurement.measurement_date >= relative_date
+            )
+            self.observation_query = self.observation_query.where(
+                Observation.observation_date >= relative_date
+            )
+            self.condition_query = self.condition_query.where(
+                ConditionOccurrence.condition_start_date >= relative_date
+            )
+            self.drug_query = self.drug_query.where(
+                DrugExposure.drug_exposure_start_date >= relative_date
+            )
+        else:
+            self.measurement_query = self.measurement_query.where(
+                Measurement.measurement_date <= relative_date
+            )
+            self.observation_query = self.observation_query.where(
+                Observation.observation_date <= relative_date
+            )
+            self.condition_query = self.condition_query.where(
+                ConditionOccurrence.condition_start_date <= relative_date
+            )
+            self.drug_query = self.drug_query.where(
+                DrugExposure.drug_exposure_start_date <= relative_date
+            )
+        return self
+
+    def add_numeric_range(
+        self,
+        min_value: float = None,
+        max_value: float = None
```

Comment:
```suggestion
        min_value: float | None = None,
        max_value: float | None = None
```

These can't be of type `float` and default to `None`